src/components/tooltip/CallNode.js

```diff
@@ -57,15 +62,241 @@ type Props = {|
+            subCategory: subcategory,
+            selfTime: selfTimeValue,
+            totalTime: self,
+          });
+        });
+      }
+    );
+
+    return (
+      <div className="tooltipCallNodeCategory">
+        {/* grid row -------------------------------------------------- */}
+        <div />
+        <div className="tooltipCallNodeHeader" />
+        <div className="tooltipCallNodeHeader">
+          <span className="tooltipCallNodeHeaderSwatchRunning" />
+          Running
+        </div>
+        <div className="tooltipCallNodeHeader">
+          <span className="tooltipCallNodeHeaderSwatchSelf" />
+          Self
+        </div>
+        {/* grid row -------------------------------------------------- */}
+        <div className="tooltipLabel tooltipCategoryLabel tooltipCategoryRowHeader">
+          Overall
+        </div>
+        <div className="tooltipCallNodeGraph tooltipCategoryRow tooltipCategoryRowHeader">
+          <div
+            className="tooltipCallNodeCategoryGraphRunning"
+            style={{
+              width: GRAPH_WIDTH,
+            }}
+          />
+          <div
+            className="tooltipCallNodeCategoryGraphSelf"
+            style={{
+              width: (GRAPH_WIDTH * selfTime.value) / totalTime.value,
+            }}
+          />
+        </div>
+        <div className="tooltipCallNodeTiming tooltipCategoryRow tooltipCategoryRowHeader">
+          {formatCallNodeNumberWithUnit(
+            weightType,
+            isHighPrecision,
+            totalTime.value
+          )}
+        </div>
+        <div className="tooltipCallNodeTiming tooltipCategoryRow tooltipCategoryRowHeader">
+          {selfTime.value === 0
+            ? '—'
+            : formatCallNodeNumberWithUnit(
+                weightType,
+                isHighPrecision,
+                selfTime.value
+              )}
+        </div>
+        {categoriesAndTime.map((entry, index) => {
+          const categoryColor = categories[entry.category].color;
+          return (
+            <React.Fragment key={index}>
+              <div
+                className={classNames({
+                  tooltipCallNodeName: true,
+                  tooltipLabel: true,
+                  tooltipCategoryLabel: entry.subCategory === -1,
+                  tooltipCategoryRowHeader: entry.subCategory === -1,
+                })}
+              >
+                {getLastCategoryPartLabel(
+                  categories,
+                  entry.categoryName,
+                  entry.category,
+                  entry.subCategory
+                )}
+              </div>
+              <div
+                className={
+                  'tooltipCallNodeGraph ' +
+                  (entry.subCategory === -1 ? 'tooltipCategoryRow' : '')
+                }
+                style={{
+                  '--running-color': `var(--category-color-${categoryColor})`,
+                  '--self-color': `var(--category-color-${categoryColor})`,
```

Comment:
> The variables look unnecessary to me and bring unneeded complexity to the CSS, instead this could be specified as background-color directly in the style entry for each bar.

Good catch. I like your first version more.